CODINGSTYLE.md

```diff
@@ -0,0 +1,443 @@
+
+### Method definition
+
+All public and protected methods must have an explicit result type.
+
+Procedure syntax must not be used. `: Unit =` must be used instead.
+
+Side-effect-free methods without formal parameters should be declared without `()`, unless either a) it overrides a methode defined with `()` (such as `toString()`) or b) it implements a Java method in the Java libraries.
+
+### for comprehensions
+
+For-comprehensions may only use `()` if they have a single generator without `if`, such as:
+
+```scala
+for (i <- 0 until n)
+  doStuff(i)
+```
+
+Otherwise, it must use `{}`, and there must be one generator per line.
+Guards (`if`s) may be either on the same line as a generator, or on a dedicated line.
+The body of the `for` is then automatically surrounding by braces too, even if it is a one-liner:
+
+```scala
+for {
+  i <- 0 until n
+  j <- 0 until i
+} {
+  doStuff(i, j)
+}
+```
+
+The `yield` keyword should be placed at the end of the `for` line, followed by braces:
+
+```scala
+for (x <- xs) yield {
+  x * 2
+}
+```
+
+For short `yield` blocks, the following can be used instead:
+
+```scala
+for (x <- xs)
+  yield x * 2
+```
+
+### ScalaDoc
+
+ScalaDoc comments that fit in one line must be written as
+
+```scala
+/** Returns the maximum of a and b. */
+def max(a: Int, b: Int): Int = ???
+```
+
+Multi-line ScalaDoc comments must use the following style:
+
+```scala
+/** Returns the maximum of a and b.
+ *
+ *  If a > b, returns a. Otherwise returns b.
+ */
+def max(a: Int, b: Int): Int = ???
+```
+
+### Non-ScalaDoc comments
+
+Normal comments fitting on one-line should use `//`.
+A comment that does not fit on one line should use the multi-line comment syntax and follow this style:
+
+```scala
+/* This complicated algorithm computes the maximum of two integer values a
+ * and b. If a > b, it computes a, otherwise it computes b.
+ */
+```
+
+
+## Class declaration
+
+A class declaration, together with its constructor parameters, its `extends` clause, and its self type, is technically a single line.
+Example:
+
+```scala
+class Foo(val x: Int) extends Bar with Foobar { self =>
+```
+
+However, this tends to become too long in many cases.
+
+If the declaration does not fit on one line, the self type must be on dedicated line, indented 2 spaces only, and followed by a blank line:
+
+```scala
+class Foo(val x: Int) extends Bar with Foobar {
+  self =>
+
+  // declarations start here
+```
+
+If too long in itself, the list of constructor parameters should be broken similarly to formal parameters to a method, i.e., indented 4 spaces, and followed by a blank line:
+
+```scala
+class Foo(val x: Int, val y: Int,
+    val z: Int) extends Bar with Foobar {
+
+  // declarations start here
+```
+
+As an exception, if the constructor parameters are a (long) list of "configuration" parameters, the following format should be used instead:
+
+```scala
+class Foo(
+    val x: Int,
```

Comment:
Picky: Some people like lining up the types/names eg :

```scala
class Foo(
    val xMassTree: Int,
    val yMe:       Int,
    val z:         Int
) extends Bar with Foobar {
```

As I'm pretty sure you _don't_ want this, if you used names of varying length you would highlight your true intent.